### DISCUSSION TITLE
How the PnP - Search Filters' filters values for the comboboxes will be consructed ?

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [microsoft-search/pnp-modern-search#2816] [Q&A] [answered]
I have added the following filters inside my PnP - Search Filters web part:-

![image](https://user-images.githubusercontent.com/79726511/222289607-82e64ad4-32e2-4ff9-98fa-7f8450fc7fba.png)

and i connect it with Search Result:-

![image](https://user-images.githubusercontent.com/79726511/222289705-aeca7abd-86eb-4751-98b6-bba6503724fb.png)

now i am not sure how the options inside the filters' combo boxes get constructed? will they cover all the options from all the related search result items/documents? or only from the ones that are going to be loaded as the user do paging through the search results?
For example; so let say the search result contain 10,000 documents, and the last document contain a option named "HR Document" for a RefinableString00 , then will the RefinableString00 contain the "HR Dcoument" option when the PnP -Search Filter web part first load the first 10 search results?

Thanks

**Comment by gretchunkim** [comment, marked as answer]
You can tell how it's constructing options by looking at Debug layout option mode on the results web part. 

Short answer, it only creates filters (dropdown options) based on the results (display) that's connected to the filter.

**Reply by kasperbolarsen** [reply]
Please see this related question, https://github.com/microsoft-search/pnp-modern-search/discussions/2718
